frontend/src/components/UserProfilePanels/UserInfo.js

```diff
@@ -1,70 +1,140 @@
-import React from 'react';
+import React, { useState } from 'react';
+import capitalize from 'lodash/capitalize';
+import trim from 'lodash/trim';
 import profilePic from '../../assets/profilePic.PNG';
 import './UserInfo.css';
+import { editCurrentUserName } from '../../lib/apiClient';
+import checkmark from '../../assets/checkmark-iconSave.png';
 
-const UserInfo = ({ user }) => (
-  <div className="row">
-    <div className="col-md-12 profile-content">
-      <div className="row profilePage-user">
-        <div className="col-md-2" align="center">
-          <img
-            src={profilePic}
-            alt="User"
-            className="profilePage-profilePic"
+function UserInfo({ user, setUser }) {
+  const [editBox, hideEditBox] = useState(false);
```

Comment:
I think the name of the variable `editBox` is kind of misleading, as it implies its value is something more complicated than a boolean (like an edit box).

Since the value of `editBox` is a boolean, indicating whether or not we are editing, I think this would make more sense:
```
const [isEditing, setIsEditing] = useState(false);
```